File: Higher or Lower Game.py
# imports the random module
import random


# defines the main procedure
def play():

    go = 'True'

    # generates the first random number, requests user input and validates it
    integer = random.randint(1, 100)
    print(f"The number is {integer}. Higher (H) or Lower (L)?")
    userInp = input("> ").upper()
    while not(userInp == 'H' or userInp == 'L'):
        print("That was not a correct input, please enter 'H' for Higher or 'L' for Lower")
        userInp = input("> ").upper()

    # loop until user loses
    while go == 'True':

        # generate a new random number
        newInteger = random.randint(1, 100)

        # check if the new number being higher or lower matches user input
        if (userInp == 'H' and newInteger > integer) or (userInp == 'L' and newInteger < integer):
            print(f"Correct! The number was {newInteger}. Higher (H) or Lower (L)?")
            userInp = input("> ").upper()
            while not(userInp == 'H' or userInp == 'L'):
                print("That was not a correct input, please enter 'H' for Higher or 'L' for Lower")
                userInp = input("> ").upper()

            # set new number to old variable in order to repeat cycle
            integer = newInteger    
        
        # end game if user input does not meet conditions, and ask user if they want to play again
        else:
            go = 'False'
            print("You lost the game!\nWould you like to play again?")
            answer = input("> ").upper()
            while not(answer == 'Y' or answer == 'N'):  
                print("It seems what you entered is incorrect. Please re-enter.\nWould you like to play again? Enter Yes (Y) or No (N) to play...")
                answer = input("> ").upper()
            else:
                if answer == 'Y':
                    play()
                elif answer == 'N':
                    print("See you later!")   


# defines main menu procedure which asks user if they want to play
def mainMenu():
    print("Hey! Welcome to Higher or Lower!\nEnter Yes (Y) or No (N) to play...")
    answer = input("> ").upper()
    while not(answer == 'Y' or answer == 'N'):  
        print("It seems what you entered is incorrect. Please re-enter.\nEnter Yes (Y) or No (N) to play...")
        answer = input("> ").upper()
    else:
        if answer == 'Y':
            play()
        elif answer == 'N':
            print("See you later!")


# call the main menu procedure, therefore beginning game
mainMenu()        

File: Higher or Lower Game.js
// import package to read input from console
const prompt = require('prompt-sync')();

// defines the main procedure
const play = () => {

    let go = true

    // generates the first random number, requests user input and validates it
    let integer = Math.floor(Math.random() * 101)
    console.log(`The number is ${integer}. Higher (H) or Lower (L)?`)
    let userInp = prompt('> ').toUpperCase()
    while (userInp != 'H' && userInp != 'L') {
        console.log("That was not a correct input, please enter 'H' for Higher or 'L' for Lower")
        userInp = prompt('> ').toUpperCase()
    };

    // loop until user loses
    while (go === true) {

        // generate a new random number
        let newInteger = Math.floor(Math.random() * 101)

        // check if the new number being higher or lower matches user input
        if ((userInp == 'H' && newInteger > integer) || (userInp == 'L' && newInteger < integer)) {
            console.log(`Correct! The number was ${newInteger}. Higher (H) or Lower (L)?`)
            userInp = prompt("> ").toUpperCase()
            while (userInp != 'H' && userInp != 'L') {
                console.log("That was not a correct input, please enter 'H' for Higher or 'L' for Lower")
                userInp = prompt('> ').toUpperCase()
            };

            // set new number to old variable in order to repeat cycle
            integer = newInteger
        }

        // end game if user input does not meet conditions, and ask user if they want to play again
        else {
            go = false
            console.log('You lost the game!\nWould you like to play again?')
            let answer = prompt("> ").toUpperCase()
            while (answer != 'Y' && answer != 'N') {
                console.log("It seems what you entered is incorrect. Please re-enter.\nWould you like to play again? Enter Yes (Y) or No (N) to play...")
                answer = prompt("> ").toUpperCase()
            };

            if (answer === 'Y') {
                play();
            }

            else if (answer === 'N') {
                console.log('See you later!')
            };
        };
    };
};

// defines main menu procedure which asks user if they want to play
const mainMenu = () => {
    console.log("Hey! Welcome to Higher or Lower!\nEnter Yes (Y) or No (N) to play...")
    let answer = prompt("> ").toUpperCase()
    while (answer != 'Y' && answer != 'N') {
        console.log("It seems what you entered is incorrect. Please re-enter.\nEnter Yes (Y) or No (N) to play...")
        answer = prompt("> ").toUpperCase()
    };

    if (answer === 'Y') {
        play();
    }

    else if (answer === 'N') {
        console.log('See you later!')
    };
};

// call the main menu procedure, therefore beginning game
mainMenu();